src/test/objectstore/allocsim/ops_scraper.py

```diff
@@ -0,0 +1,305 @@
+
+    pref_freq = int(Env.args().freq)
+    freq = int(Env.args().freq)
+
+    sleep_time = 0
+    update_period = 10
+    sum_time_elapsed = 0
+    min_history_size = int(Env.args().min_history_size)
+    max_history_size = int(Env.args().max_history_size)
+    history_size = max_history_size
+    # -------------|-------------------|
+    history_overlap = 1.10
+    osds = Env.args().osds.split(',')
+    osds_info = []
+    for osd in osds:
+        _set_osd_history_size(osd, max_history_size)
+        osds_info.append(OsdParameters(osd, 0, freq, history_size))
+
+    logger.debug(f'start freq {freq}')
+    while True:
+        logger.debug(f'sleep sec {sleep_time}')
+        time.sleep(sleep_time)
+        sleep_time = freq
+        for osd in osds_info:
+
+            if osd.ready_time >= time.time():
+                continue
+            process_info = process_osd(osd.name)
+
+            if not process_info.new_ops:
+                new_ops_period = 1
+            else:
+                oldest_initiated_at = process_info.oldest_operation.initiated_at
+                new_ops_period = process_info.capture_period_end - _to_timestamp(oldest_initiated_at)
+            capture_period = process_info.capture_period_end - process_info.capture_period_start
+
+            sum_time_elapsed += capture_period
+            osd.sum_ops += process_info.new_ops
+
+            if process_info.new_ops >= max_history_size:
+                lost_ops = ((process_info.new_ops * capture_period) / (new_ops_period)) - process_info.new_ops
+                logger.debug(f'process_info.new_ops: {process_info.new_ops} capture_period: {capture_period} new_ops_period: {new_ops_period} start: {process_info.capture_period_start} end: {process_info.capture_period_end}')
+                osd.sum_ops += lost_ops
+
+
+
+
+            outfile.write(process_info.processed_info)
+            outfile.flush()
+
+            sleep_time -= process_info.process_time + process_info.command_time
+            osd.ready_time = time.time() + (osd.freq - process_info.process_time + process_info.command_time)
+            sleep_time = max(0, sleep_time)
+            logger.info(f'osd.{osd.name} parsing dump_historic_ops with {process_info.ops_count} ops took {process_info.process_time}')
+            logger.info(f'osd.{osd.name} command dump_historic_ops with {process_info.ops_count} ops took {process_info.command_time}')
+
+            osd.periods += 1
+            if (osd.periods % update_period) == 0:
+                ops_per_period = (osd.sum_ops / osd.periods)
+                new_history_size = int(ops_per_period * history_overlap)
+                if new_history_size >= min_history_size and new_history_size <= max_history_size:
+                    _set_osd_history_size(osd.name, new_history_size)
+                    logger.info(f'changing osd.{osd.name} history size from {history_size} to {new_history_size}')
+                    history_size = new_history_size
```

Comment:
I imagined that both `capture_period` and `history_size` will be a values that are adapting per-OSD.
So the shape of the main loop will be something like:
```
while true:
  sleep(time until nearest OSD needs to be captured)
  for all OSDs:
    if osd is due:
      capture
      write captured to outfile
      adapt capture params if necessary (frequency, history_size)
      recalculate new sleep time for main loop
```
Can the "adapt capture params" step be moved to separate function?